packages/core/src/odata-v4/request-builder/update-request-builder.ts

```diff
@@ -41,37 +28,14 @@ export class UpdateRequestBuilderV4<EntityT extends EntityV4>
     readonly _entityConstructor: Constructable<EntityT>,
     readonly _entity: EntityT
   ) {
-    super(new ODataUpdateRequestConfig(_entityConstructor, oDataUriV4));
-    this.requestConfig.eTag = _entity.versionIdentifier;
-    this.required = new Set<string>();
-    this.ignored = new Set<string>();
-
-    this.requestConfig.keys = oDataUriV4.getEntityKeys(
-      this._entity,
-      this._entityConstructor
-    );
-
-    this.requestConfig.payload = this.getPayload();
-  }
-
-  get entity(): EntityT {
-    return this._entity;
-  }
-
-  /**
-   * @deprecated Since v1.29.0. This method should never be called, it has severe side effects.
-   * Builds the payload and the entity keys of the query.
-   * @returns the builder itself
-   */
-  prepare(): this {
-    this.requestConfig.keys = oDataUriV4.getEntityKeys(
-      this._entity,
-      this._entityConstructor
+    super(
+      _entityConstructor,
+      _entity,
+      oDataUriV4,
+      entitySerializerV4,
+      extractODataEtagV4,
+      identity
```

Comment:
Question: why v4 is different in terms of the payload manipulation. Is it a bug or intended?

Reply:
It is intended. For v2 the complex and nav props are removed. I do not know why. Can be overloaded with requiredFields.